File: 0781-rabbits-in-forest/0781-rabbits-in-forest.cpp
class Solution {
public:
    int numRabbits(vector<int>& answers) {
        map<int, int> vfreq;
        for(auto a : answers)
            vfreq[a + 1]++;

        int ans = 0;
        for(auto [f, n] : vfreq)
            ans += (n / f + (n % f != 0)) * (f);

        return ans;
    }
};

File: 0781-rabbits-in-forest/0781-rabbits-in-forest.java
import java.util.*;

class Solution {
    public int numRabbits(int[] answers) {
        Map<Integer, Integer> freq = new HashMap<>();

        for (int a : answers) {
            freq.put(a, freq.getOrDefault(a, 0) + 1);
        }

        int count = 0;
        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            int k = entry.getKey();
            int v = entry.getValue();
            int groupSize = k + 1;
            int groups = (v + k) / groupSize;
            count += groups * groupSize;
        }

        return count;
    }
}